multilingual-polylang.php

```diff
@@ -67,10 +67,16 @@ public static function get_permalink( $post ) {
 			$post_language = pll_get_post_language( $post->ID );
 
 			if ( $post_language != self::$current_language ) {
-				$search = "/$post_language/";
-				$replace = '/' . self::$current_language . '/';
-
-				$link = str_replace( $search, $replace, $link );
+				global $polylang;
+				
+				if( $polylang->options['hide_default'] ) {
+					$add = '/' . self::$current_language;
+					$link = self::rebuild_url( $link, $add );
```

Comment:
I'm wondering if we can avoid having two different ways of rewriting the URL here. I think my original `str_replace()` is a bit sloppy and could potentially end up causing errors. On the other hand, `rebuild_url()` seems a bit heavy-handed.